daemon/firewall/firewall_manager.go

```diff
@@ -0,0 +1,316 @@
+
+	if f.enabled {
+		if err := f.commandExecutor.ExecuteCommand("-I " + inputDportRule); err != nil {
+			return fmt.Errorf("allowlisting input dport: %w", err)
+		}
+
+		if err := f.commandExecutor.ExecuteCommand("-I " + inputSportRule); err != nil {
+			return fmt.Errorf("allowlisting input sport: %w", err)
+		}
+
+		if err := f.commandExecutor.ExecuteCommand("-I " + outputDportRule); err != nil {
+			return fmt.Errorf("allowlisting output dport: %w", err)
+		}
+
+		if err := f.commandExecutor.ExecuteCommand("-I " + outputSportRule); err != nil {
+			return fmt.Errorf("allowlisting input dport: %w", err)
+		}
+	}
+
+	f.allowlistRules = append(f.allowlistRules, inputDportRule)
+	f.allowlistRules = append(f.allowlistRules, inputSportRule)
+	f.allowlistRules = append(f.allowlistRules, outputDportRule)
+	f.allowlistRules = append(f.allowlistRules, outputSportRule)
+
+	return nil
+}
+
+func (f *FirewallManager) SetAllowlist(udpPorts []int, tcpPorts []int, subnets []netip.Prefix) error {
+	ifaces, err := f.devices()
+	if err != nil {
+		return fmt.Errorf("listing interfaces: %w", err)
+	}
+
+	for _, subnet := range subnets {
+		for _, iface := range ifaces {
+			inputRule := fmt.Sprintf("INPUT -s %s -i %s -m comment --comment nordvpn -j ACCEPT", subnet.String(), iface.Name)
+			outputRule := fmt.Sprintf("OUTPUT -d %s -o %s -m comment --comment nordvpn -j ACCEPT", subnet.String(), iface.Name)
+
+			if f.enabled {
+				if err := f.commandExecutor.ExecuteCommand("-I " + inputRule); err != nil {
+					return fmt.Errorf("adding input accept rule for subnet: %w", err)
+				}
+				if err := f.commandExecutor.ExecuteCommand("-I " + outputRule); err != nil {
+					return fmt.Errorf("adding output accept rule for subnet: %w", err)
+				}
+			}
+
+			f.allowlistRules = append(f.allowlistRules, inputRule)
+			f.allowlistRules = append(f.allowlistRules, outputRule)
+		}
+	}
+
+	udpPortRanges := portsToPortRanges(udpPorts)
+	for _, portRange := range udpPortRanges {
+		for _, iface := range ifaces {
+			if err := f.allowlistPort(iface.Name, "udp", portRange); err != nil {
+				return fmt.Errorf("allowlisting udp ports: %w", err)
+			}
+		}
+	}
+
+	tcpPortRanges := portsToPortRanges(tcpPorts)
+	for _, portRange := range tcpPortRanges {
+		for _, iface := range ifaces {
+			if err := f.allowlistPort(iface.Name, "tcp", portRange); err != nil {
+				return fmt.Errorf("allowlisting tcp ports: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (f *FirewallManager) UnsetAllowlist() error {
+	if f.enabled {
+		for _, rule := range f.allowlistRules {
+			if err := f.commandExecutor.ExecuteCommand("-D " + rule); err != nil {
+				return fmt.Errorf("removing allowlist rule: %w", err)
+			}
+		}
+	}
+
+	f.allowlistRules = nil
+
+	return nil
+}
+
+func (f *FirewallManager) manageApiAllowlist(allow bool) error {
+	iptablesMode := "-I "
+	if !allow {
+		iptablesMode = "-D "
+	}
+
+	ifaces, err := f.devices()
```

Comment:
Shouldn't this also be stored into a list? If the devices list changes between add and remove then the remove will fail, leaving rules into the FW.

Reply:
Fixed, not detected by GH as outdated.